app/assets/javascripts/admin/order_cycles/services/product.js.coffee

```diff
@@ -0,0 +1,22 @@
+angular.module('admin.orderCycles').factory('Product', ($resource) ->
+  Product = $resource('/admin/exchanges/:exchange_id/products.json', {}, {
```

Comment:
yes, agree, this should be called ProductResource in this case.
In terms of the service name, I am not using byEnterprise as you suggest because I removed the internal data structure (see comment and commit below).

So, I renamed the service to something a lot more appropriate ExchangeProduct and the resource to ExchangeProductResource. New commit: d68bcb7a1cd2ce5157b99b2c8f20b06dccc6e279

Reply:
I think this point is resolved, right?